pinc/Project.inc

```diff
@@ -133,40 +137,85 @@ class Project
 
     // -------------------------------------------------------------------------
 
+    private function _get_PPer()
+    {
+        global $site_supports_corrections_after_posting;
+
+        // The logic to determine who is a PPer is a bit convoluted
+        // as it depends on the postproofer column, the checkedoutby
+        // column, and the state of the project.
+
+        $checkedout_states = [
+            PROJ_POST_SECOND_CHECKED_OUT,
+            PROJ_POST_COMPLETE,
+            PROJ_SUBMIT_PG_POSTED
+        ];
+        if ( $site_supports_corrections_after_posting )
+        {
+            $checkedout_states[] = PROJ_CORRECT_CHECKED_OUT;
+        }
+
+        $PPer = "";
+        if ( !empty($this->postproofer) )
+        {
+            $PPer = $this->postproofer;
+        }
+        else if ( !empty($this->checkedoutby) &&
+            !in_array($this->state, $checkedout_states)
+        )
+        {
+            $PPer = $this->checkedoutby;
+        }
+
+        return $PPer;
+    }
+
+    private function _get_PPVer()
+    {
+        // The logic to determine who is a PPVer is a bit convoluted
+        // as it depends on the ppverifier column, the checkedoutby
+        // column, and the state of the project.
+
+        $PPVer = "";
+        if ( !empty($this->ppverifier) )
+        {
+            $PPVer = $this->ppverifier;
+        }
+        else if ( !empty($this->checkedoutby) &&
+            in_array(
+                $this->state,
+                [
+                    PROJ_POST_SECOND_CHECKED_OUT,
+                    PROJ_POST_COMPLETE,
+                    PROJ_SUBMIT_PG_POSTED,
+                ]
```

Comment:
This says pretty much the same thing as `private function _get_PPer()`, but `_get_PPer()` assigns the acceptable states to `$checkedout_states`. Functionality is the same, and I realize that the code was just moved from one place to another, but should they follow the same pattern?

Reply:
Yes, it would be better to have them more closely match. Fixed.